### DISCUSSION TITLE
Community wide "good first issue" overview

### DISCUSSION THREAD

**Discussion by finestructure** [SwiftPackageIndex/SwiftPackageIndex-Server#1362] [Ideas]
@fabianfett suggested the SPI might be a good place to aggregate and show "starter" issues, i.e. issues in projects that are typically tagged with "good first issue".

The idea is to make it easier for people interested in contributing to open source projects to get an overview of what projects have starter issues and more easily browse them.

**Comment by daveverwer** [comment]
This ties in well with some of what we've been discussing for the home page (which there isn't an issue for yet). I also want to tag #852 here as we start to bring some of this together.

Thanks @fabianfett!

**Comment by fabianfett** [comment]
What would be the next steps from your side to move this forward?

**Comment by daveverwer** [comment]
That’s a good question, and the lack of a concrete plan is why we moved it from being an issue to being a discussion.

The obvious place for a feature like this to live would be on the home page. You want to catch people’s eyes as they browse and maybe consider contributing to a project. We do have a re-design of the home page planned, and this would fit in well with that re-design, but that task isn’t imminent.

I wonder if there’s a way we can prototype the idea without building software? Maybe you could propose a Swift forum where package authors can promote issues that fit this criteria? If that proves popular, it could even be a source of this feature over and above them tagging issues with the `good-first-issue` label, which sounds good in principle, but might be tricky to curate across 4,000 packages.

I know that’s a complete cop-out answer, but we’re certainly going to need some help making this a reality if it's to be an effective way to foster collaboration on open-source Swift projects.

**Comment by Sherlouk** [comment]
I might be completely off base but could part of this be to calculate "response times" - how long pull requests remain untouched for, etc?

Being new to OSS, you want communities which are active and willing to partake in conversations and provide feedback.

Repos may have lots of "good first issue" tickets but aren't being maintained. I don't think these are great candidates for us to surface.
